src/gwt/src/org/rstudio/studio/client/workbench/views/connections/model/NewConnectionContext.java

```diff
@@ -17,18 +17,60 @@
 
 
 import java.util.ArrayList;
-import java.util.List;
-
-import org.rstudio.core.client.StringUtil;
-import org.rstudio.studio.client.application.Desktop;
 
 import com.google.gwt.core.client.JavaScriptObject;
-import com.google.gwt.core.client.JsArray;
-import com.google.gwt.core.client.JsArrayString;
 
 public class NewConnectionContext extends JavaScriptObject
 {
    protected NewConnectionContext()
    {
    }
+
+   public static class NewConnectionInfo extends JavaScriptObject
+   {
+      protected NewConnectionInfo()
+      {
+      }
+
+      public final native String getPackage() /*-{
+         return this["package"];
+      }-*/;
+
+      public final native String getName() /*-{
+         return this["name"];
+      }-*/;
+
+      public final native String getType() /*-{
+      	 return this["type"];
+      }-*/;
+   }
+
+   public final native int getConnectionsLength() /*-{
```

Comment:
Nit: You could expose this more idiomatically as a `JsArray<NewConnectionInfo>`, which would also allow you to use e.g. `JsArrayUtil.toArrayList` below.